lib/openapi_first/middlewares/response_validation.rb

```diff
@@ -24,13 +24,21 @@ def initialize(app, options = {})
       def call(env)
         request = find_request(env)
         status, headers, body = @app.call(env)
-        body = body.to_ary if body.respond_to?(:to_ary)
+        body = read_body(body)
         request.validate_response(Rack::Response[status, headers, body], raise_error: true)
         [status, headers, body]
       end
 
       private
 
+      def read_body(body)
+        return body.to_ary if body.respond_to?(:to_ary)
+
+        result = []
+        body.each { |part| result << part }
+        result
```

Comment:
nice one!

dropping this snippet here as a fun fact - ruby has a dedicated method for these :)

`body.each_with_object([]) { |part, result| result << part }`

Reply:
I am a bis hesitant about using something else that #each or #to_ary here because the Rack spec explicitly mentions #each and not Enumerable

Reply:
Oh, you are absolutely right!